x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key_so.ts

```diff
@@ -112,7 +115,50 @@ export async function generateEnrollmentAPIKey(
   }
   const agentPolicyId =
     data.agentPolicyId ?? (await agentPolicyService.getDefaultAgentPolicyId(soClient));
+
+  if (providedKeyName) {
+    let hasMore = true;
+    let page = 1;
+    let keys: EnrollmentAPIKey[] = [];
+    while (hasMore) {
+      const { items } = await listEnrollmentApiKeys(soClient, {
+        page: page++,
+        perPage: 100,
+        kuery: `${ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE}.policy_id:${agentPolicyId} and ${ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE}.name:${providedKeyName.replace(
```

Comment:
I am wondering if we could have something like `name:"test*"` it could make the escaping easier

Reply:
In my testing, quotes and wildcards didn't seem to work together